packages/wrangler/src/dialogs.ts

```diff
@@ -0,0 +1,96 @@
+import assert from "assert";
+import chalk from "chalk";
+import prompts from "prompts";
+import { CI } from "./is-ci";
+import isInteractive from "./is-interactive";
+import { logger } from "./logger";
+
+// TODO: Use this function across the codebase.
+function isNonInteractiveOrCI(): boolean {
+	return !isInteractive() || CI.isCI();
+}
+
+export async function confirm(
+	text: string,
+	defaultValue = true
+): Promise<boolean> {
+	if (isNonInteractiveOrCI()) return defaultValue;
+	const { value } = await prompts({
+		type: "confirm",
+		name: "value",
+		message: text,
+		initial: defaultValue,
+	});
+	return value;
+}
+interface PromptOptions {
+	defaultValue?: string;
+	isSecret?: boolean;
+}
+
+export async function prompt(
+	text: string,
+	options: PromptOptions = {}
+): Promise<string> {
+	if (isNonInteractiveOrCI()) {
+		assert(
+			options?.defaultValue !== undefined,
+			"A default value must be provided in non-interactive contexts"
+		);
+		return options.defaultValue;
+	}
+	const { value } = await prompts({
+		type: "text",
+		name: "value",
+		message: text,
+		initial: options?.defaultValue,
+		style: options?.isSecret ? "password" : "default",
+	});
+	return value;
+}
+
+interface SelectOption<Values> {
```

Comment:
This is mostly for consistency with the previous type definitions, and I’m not certain if prompts requires it to be a string. That said, since it was a string in the ink based definitions, I think it’s worth keeping it that here unless there’s a specific use case where widening it is needed.